src/services/facilityService.ts

```diff
@@ -41,12 +59,41 @@ export const searchFacilities = async (userSearchQuery : string[]) : Promise<Fac
 
 const mapDbEntityTogqlEntity = (dbEntity : DocumentData) : Facility => {
     const gqlEntity = {
-        id: dbEntity.id,
         nameEn: dbEntity.nameEn,
         nameJa: dbEntity.nameJa,
         contact: dbEntity.contact,
-        healthcareProfessionals: dbEntity.healthcareProfessionals
+        healthcareProfessionalIds: dbEntity.healthcareProfessionals
     } satisfies Facility
     
     return gqlEntity
 }
+
+function validateContactInput(contactInput: Contact) : Contact {
+    const facilityContact = {
+        address: contactInput.address as PhysicalAddress,
+        email: contactInput.email as string,
+        mapsLink: contactInput.mapsLink as string,
+        phone: contactInput.phone as string,
+        website: contactInput.website as string
+    }
+
+    return facilityContact
+}
+
+function validateNameEnInput(nameEnInput: string) : string {
+    const nameEn = nameEnInput
+
+    return nameEn
+}
+
+function validateNameJaInput(nameJaInput: string) : string {
+    const nameJa = nameJaInput
+
+    return nameJa
+}
+
+function mapAndValidateHealthcareProInput(healthcareProInput: HealthcareProfessionalInput[]) : Promise<string[]> {
```

Comment:
since we have a healthcare professional service, we should move this there, and call it from here. (also the dependent sub methods as well)